modules/ui/src/main/scala/lucuma/ui/optics/ValidFormatNecInstances.scala

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.optics
+
+import cats.Order
+import cats.data.NonEmptyChain
+import cats.data.ValidatedNec
+import cats.syntax.all._
+import eu.timepit.refined.types.string.NonEmptyString
+import eu.timepit.refined.auto._
+
+trait ValidFormatNecInstances {
```

Comment:
These are useful for using `composeValidFormat` on a ValidFormatInput when one field depends on another.

I fixed the error type to `NonEmptyString` because it makes it much easier to supply the error in user code.  ValidFormatNec doesn't fix the error type in general, but it still seemed like this was a reasonable place to put them. I'm happy to entertain alternatives, though.